core/shared/src/main/scala/org/scalacheck/Prop.scala

```diff
@@ -194,7 +194,7 @@ object Prop {
       status: Status,
       args: List[Arg[Any]] = Nil,
       collected: Set[Any] = Set.empty,
-      labels: Set[String] = Set.empty
+      labels: Set[() => String] = Set.empty
```

Comment:
@Daenyth pointed out on discord that `Set[() => String]` could be problematic as the labels aren't deduped as they're added. They'll still be deduped before rendering when [`map`ped to a `Set[String]`](https://github.com/typelevel/scalacheck/pull/979/files#diff-b3c478fbc4749171669125a61d9976e61a7f64aea71ecba0f6667e09cb0c97c4R157), but I wanted to call it out here in case anyone thinks a different collection type would be better